bin/zephir

```diff
@@ -19,9 +19,8 @@ if [ -z "$ZEPHIRDIR" ]; then
   fi
 fi
 
-if [[ $1 && $2 && $3 && "$1"=="-c" ]]; then
+if [ $1 -a $2 -a $3 -a $1 = "-c" ]; then
```

Comment:
just a guess, but maybe your version of test has some of those variables empty and is essentially running `test -a`

maybe adding a ' around each variable would work for you?
`if [ '$1' -a '$2' -a '$3' -a '$1' = "-c" ]; then`
I haven't had a chance to check this yet though